astropy/stats/funcs.py

```diff
@@ -1185,7 +1189,7 @@ def func(s: float) -> float:
         out = eqn9_left(s_min, s, N, B)
         return out[0] - CL
 
-    S_max = brentq(func, N - B, 100)
+    S_max = brentq(func, N - B, max(100, np.sqrt(N - B) * 10 + N))
```

Comment:
Where does the `np.sqrt(N - B) * 10 + N)` term come from?  Any reference?

Reply:
It was made up thinking about the Poisson distribution, i.e., something like `10 sigma` over the input number, but ensuring it doesn't become too low using the `max`. 

Not sure how to give something that will also work if someone asks for a 20 sigma result... I guess at some level one wants to put infinity there, or rather the largest number that will work, then one should probably work in `log(N)` in the first place.